src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/ReflectMethods.java

```diff
@@ -736,16 +736,15 @@ public Value toValue(JCTree.JCStatement statement) {
         }
 
         Value coerce(Value sourceValue, Type sourceType, Type targetType) {
-            if (targetType.hasTag(TypeTag.VOID)) {
-                // if target type is void, nothing to coerce
+            if (targetType == null || targetType.hasTag(TypeTag.VOID)) {
```

Comment:
Not sure about this -- as I'm not sure about `VOID`. E.g. this method should be called when we do have a target type. If there's no target type then there's nothing to coerce in the first place, and we shouldn't call this method. Can you describe more precisely why we end up with a `null` here?